### PR TITLE
CVSB-10752 - make tech records return an array of results from querie…

### PR DESCRIPTION
**Pull request by KHodderDVSA**
…s, and handle systemNumber based queries

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -193,6 +195,7 @@ resources:
               NonKeyAttributes:
                 - primaryVrm
                 - secondaryVrms
+                - systemNumber
```
Comment (m-ujaffer) [inline serverless.yml]: should this be a part of both key schema as well as non key attributes?

File: package.json [hunk 2]
```diff
@@ -18,50 +18,50 @@
     "prepush": "BRANCH=local npm test && npm run tslint",
     "predeploy": "npm install && npm run prepush",
     "security-checks": "git secrets --scan && git log -p | scanrepo",
-    "tslint": "tslint src/**/*.ts tests/**/*.ts -q",
+    "tslint": "tslint src/**/*.ts tests/**/*.ts -q --fix",
```
Comment (agoncearenco) [inline package.json]: Was this change intended? I think this will let sonar-scanner pass even if the committed code has listing issues

File: tests/unit/TechRecordsService.unitTest.ts [hunk 3]
```diff
@@ -195,31 +195,31 @@ describe("getTechRecordsList", () => {
...
-      try {
-        expect(await techRecordsService.getTechRecordsList("", "", SEARCHCRITERIA.ALL)).toThrowError();
-      } catch (errorResponse) {
-        expect(errorResponse).toBeInstanceOf(HTTPError);
-        expect(errorResponse.statusCode).toEqual(422);
-        expect(errorResponse.body).toEqual(HTTPRESPONSE.MORE_THAN_ONE_MATCH);
-      }
-    });
-  });
+  // context("when db returns too many results", () => {
```
Comment (m-ujaffer) [inline tests/unit/TechRecordsService.unitTest.ts]: can we remove all commented code?

File: tests/features/step_definitions/4924.ACs.feature.steps.intTest.ts [hunk 4]
```diff
@@ -32,71 +32,71 @@ defineFeature(feature, test => {
...
-    });
-    and('multiple results found (more than one CompleteTechRecord object is returned)', () => {
-    });
-    then('the system returns an HTTP status code 422', () => {
-      expect(response.status).toEqual(422);
-    });
-    ctx.succeed('done');
-    ctx = null;
-  });
+  // test("AC3 Multiple results returned", ({ given, when, then, and }) => {
```
Comment (m-ujaffer) [inline tests/features/step_definitions/4924.ACs.feature.steps.intTest.ts]: Is this needed? It can be removed.

File: serverless.yml (hunk 1 above)
Comment (agoncearenco) [reply serverless.yml]: I had the same question. I think is fine because it's a non-key attribute on `GlobalSecondaryIndexes` (hidden by the GitHub interface)

File: tests/integration/techRecords.intTest.ts [hunk 5]
```diff
@@ -74,12 +74,12 @@ describe("techRecords", () => {
 
       context("and the vehicle has more than one tech record", () => {
```
Comment (KHodderDVSA) [inline tests/integration/techRecords.intTest.ts]: Need to add a test which gets back multiple results

File: serverless.yml (hunk 1 above)
Comment (KHodderDVSA) [reply serverless.yml]: Yeah, systemNumber is a Local key, but on the GSIs, they have their own index keys. These NonKeyAttributes effectively determine which fields we get back (apparently)

File: package.json (hunk 2 above)
Comment (KHodderDVSA) [reply package.json]: Nope. Taking it out
